cvat/rq_patching.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2023 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import traceback
+from datetime import datetime
+from typing import Optional
+
+import rq.registry
+from rq.exceptions import AbandonedJobError, NoSuchJobError
+from rq.job import JobStatus
+from rq.utils import current_timestamp
+
+
+# NOTE: we should patch implementation of original method because
+# there is no enqueuing dependent jobs in original function
+# https://github.com/rq/rq/issues/2006
```

Comment:
Since you already have a fix, maybe you could submit it upstream? That way we could potentially remove this hack later (or maybe not apply it in the first place, depending on how quickly the maintainers respond).